core/drivers/crypto/caam/include/caam_utils_mem.h

```diff
@@ -0,0 +1,114 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   Memory management utilities.
+ *         Primitive to allocate, free memory.
+ */
+
+#ifndef __CAAM_UTILS_MEM_H__
+#define __CAAM_UTILS_MEM_H__
+
+#include <caam_common.h>
+
+/*
+ * Allocate normal memory.
+ * Depending on the MEM_TYPE_DEFAULT value initialized it to 0's.
```

Comment:
I still think this is not nice. Looks like the API is not relaiable and can evolve.
I would strongly suggest that `MEM_TYPE_DEFAULT` relates to an immutable description.

Reply:
ok. I'll change to define X_calloc_Y functions to make thing clearer and remove the MEM_TYPE_DEFAULT